handbook.md

```diff
@@ -0,0 +1,233 @@
+* Use ```:success, :not_authorized, etc...```, instead of actual HTTP status codes.
+* Favor FactoryGirl over fixtures. Only use fixtures when the object is created very often throughout the test suite.
+* If your tests are repeating code except for a couple of changes, wrap it in a `context` block, or write a helper that takes in the changes through arguments and generates the tests.
+* Use stubs and mocks extensively to test external calls, slow calls not related with the actual test, and also to avoid testing framework features. They make our test base much faster.
+* Use `setup_user` if you need to create an user with special permissions and roles instead of doing it yourself in the test.
+* For functional tests, if several controllers share the same behavior, extract it to a [shared functional concern](https://github.com/theforeman/foreman/blob/develop/test/functional/shared/report_host_permissions_test.rb)
+
+
+### Strings and translations
+
+Be mindful of our [Translating](http://projects.theforeman.org/projects/foreman/wiki/Translating) guide. A quick overview:
+
+#### Ruby
+
+*    To translate a string use `_('My string')`
+*    To translate string with a parameter use `_("String with param: %s") % param`
+*    To translate string with more than one parameters do not use `_("Params: %s and %s") % [param1, param2]` which is translator-unfriendly
+*    Therefore for more than one parameters use `_("Params: %{a} and %{b}") % {:a => foo, :b => bar}`
+*    Note that parameters are not translated. Variables should not contain text that needs translation.
+*    To mark something for translation (but not translate) use `N_("String")`
+*    To use plural form use `n_("One", "Two", number)` - note this function always accepts three parameters as the base language is usually English but translators are able to define as many plural forms as they need.
+*    Plural forms are usually used with one parameter, do not forget to add trailing parameter to it: `n_("%s minute", "%s minutes", @param) % @param`
+*    When using strings with ERB (e.g. for deface gem), escape properly: `n_('Test <%%= %{var1} %%>') % { :var1 => "xxx" }`
+*    Note that gettext does not extract from interpolation - this will not work: `"blah #{_('key')} blah"`
+
+#### JavaScript
+
+* Both `__` and `n__` functions are available and work in much the same way as in Ruby, with the following exceptions..
+* Interpolation of values uses `Jed.sprintf` on the translated string, so for a single parameter: `Jed.sprintf(__("Foo: %s"), "bar")`
+* Multiple parameters must be named: `Jed.sprintf(__("Example: %(foo) %(bar)"), {foo: "a", bar: "b"}) `
+
+## Code reviews
+
+We review pull requests on GitHub. Most projects have one or two maintainers that review pull requests. Everyone is invited to review code.
+
+### How to become a reviewer
+
+Start reviewing patches. Simple as that.
```

Comment:
I also think, this is slightly flippant, compared to the rest of the doc - it makes it sound unimportant, at least to me. Also, while people *should* read the whole doc, they may skip bits, so some slight repetation of key facts is useful I think. Can we flesh it out a bit? Something like
```
Simply start reviewing patches - as detailed above, all patches are available publicly as GitHub Pull Requests. Use the guidlelines below to assist you in your review, and don't be afraid to question. We all started out not knowing the code, and reviews are a good way to learn why things are done a certain way.